crates/core/tedge_agent/src/device_profile_manager/mod.rs

```diff
@@ -40,3 +49,10 @@ action = "cleanup"
 
 [failed]
 action = "cleanup"
+"#;
+
+            create_file_with_defaults(workflow_file, Some(workflow_definition))?;
```

Comment:
Creating a toml file for a builtin workflow is an idea to explore. Indeed, this would simplify the logic (with all the workflow definitions being file based) and this would give a pattern for the advanced user willing to tune an operation workflow.